src/main/java/ch/uzh/ifi/fs22/sel/group15/cozycave/server/controller/ListingController.java

```diff
@@ -48,30 +44,169 @@ public class ListingController {
     }
 
     // Get all listings in a list
-    @GetMapping("/listings")
+   /*@GetMapping("/listings")
     @ResponseStatus(HttpStatus.OK)
     @ResponseBody
     public List<ListingGetDto> getAllListings() {
         return listingService.getListings().stream()
             .map(ListingMapper.INSTANCE::listingToListingGetDto)
             .collect(Collectors.toList());
+    }*/
+    @GetMapping("/listings")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public List<ListingGetDto> getAllListingsFiltered(
+            @RequestParam(name = "MIN_RENT") Optional<String> minRent,
+            @RequestParam(name = "MAX_RENT") Optional<String> maxRent,
+            @RequestParam(name = "AVAILABLE_TO") Optional<List<String>> availableTo,
+            @RequestParam(name = "LISTING_TYPE") Optional<List<String>> listingType,
+            @RequestParam(name = "CITY") Optional<String> city,
+            @RequestParam(name = "ZIP_CODE") Optional<String> zipCode,
+            @RequestParam(name = "MIN_SQM") Optional<String> minSqm,
+            @RequestParam(name = "MAX_SQM") Optional<String> maxSqm,
+            @RequestParam(name = "AVAILABLE") Optional<String> available,
+            @RequestParam(name = "SORT") Optional<String> sort,
+            @RequestParam(name = "ORDER") Optional<String> order
+    ) {
+        HashMap<String, Object> filtersMap = new HashMap<String, Object>();
+        List<Listing> allListings = listingService.getListings();
+
+        // check if values are filled
+        if (minRent.isPresent()) {
+            try {
+                filtersMap.put("MIN_RENT", Integer.parseInt(minRent.get()));
+            } catch (NumberFormatException e) {
+                throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Expected a number");
```

Comment:
have you tried to use Integer instead in arg declaration? like: `@RequestParam(name = "MIN_RENT") Optional<Integer> minRent`? this could result in less code because we could drop some checks

Reply:
oh indeed, that's way smarter